src/main/java/com/github/maven_nar/NarCompileMojo.java

```diff
@@ -177,6 +177,22 @@ private void createLibrary(final Project antProject, final Library library)
       }
     }
 
+      if(getOS().equals( OS.WINDOWS ) && getArchitecture().equals("amd64"))
+      {
+
+          int noOfASMSources = getSourcesFor(getAssembler()).size();
+
+          if(noOfASMSources > 0)
+          {  // Assmbler files exist
+
```

Comment:
Since it is rare cases where ASM files will exist, and compiled, and linked with other C/C++ objects - There is no need every time to add the relevant Assembler compiler, only if the actual files exist